clang/lib/AST/Decl.cpp

```diff
@@ -1593,6 +1594,11 @@ void NamedDecl::printNestedNameSpecifier(raw_ostream &OS,
     Ctx = Ctx->getParent();
   }
 
+  if (Contexts.empty() && WithGlobalNsPrefix) {
```

Comment:
What does this handle here?  If Contexts is empty, don't we know that it already was a TU decl?  Is the contexts.empty test necessary at all?  Should we just always print the global NS prefix (under the flag)?

It would seem to me that the last element of Contexts will ALWAYS have a parent of TU.

What I'm saying is, Couldn't this be simplifiied to (and removing the change below):

if (WithGlobalNsPrefix) OS << "::";

Reply:
I'm not sure. The loop seems to emit things like 'anonymous namespaces'. I didn't think we wanted this before that so I added it only in cases you don't enter the loop. And inside the loop only for namespaces. I can try changing this and see if anything breaks

Reply:
Done